hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -162,9 +163,13 @@ public List<HoodieRollbackRequest> getRollbackRequests(HoodieInstant instantToRo
                     listBaseFilesToBeDeleted(instantToRollback.requestedTime(), baseFileExtension, partitionPath, metaClient.getStorage())));
               } else {
                 // if this is part of a restore operation, we should rollback/delete entire file slice.
-                hoodieRollbackRequests.addAll(getHoodieRollbackRequests(partitionPath,
-                    listAllFilesSinceCommit(instantToRollback.requestedTime(), baseFileExtension, partitionPath,
-                        metaClient)));
+                if (metaClient.getTableConfig().getTableVersion().lesserThan(HoodieTableVersion.EIGHT)) {
+                  hoodieRollbackRequests.addAll(getHoodieRollbackRequests(partitionPath, filesToDelete.get()));
+                } else {
+                  hoodieRollbackRequests.addAll(getHoodieRollbackRequests(partitionPath,
```

Comment:
can we add java docs whats differerence b/w version 6 and 8

Reply:
Addressed